File: backend/app/schemas/admin_panel/payments.py
"""Admin Payments schemas — global listing."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class AdminPaymentListItem(BaseModel):
    id: int
    user_id: int
    user_tg_id: int | None = None
    user_username: str | None = None
    subscription_id: int | None = None
    purpose: str
    provider: str
    external_id: str | None = None
    amount_kop: int
    currency: str
    status: str
    created_at: datetime
    paid_at: datetime | None = None


class AdminPaymentsPage(BaseModel):
    items: list[AdminPaymentListItem]
    total: int
    page: int
    page_size: int

File: admin/src/types/payments.ts
/**
 * Global payments admin DTOs.
 *
 * Mirrors backend.app.schemas.admin_panel.payments.
 */

export type PaymentStatus = 'pending' | 'paid' | 'failed' | 'expired' | 'refunded';
export type PaymentProvider =
  | 'platega_sbp'
  | 'platega_crypto'
  | 'cryptobot'
  | 'balance';
export type PaymentPurpose = 'subscription' | 'topup';

export interface AdminPaymentListItem {
  id: number;
  user_id: number;
  user_tg_id: number | null;
  user_username: string | null;
  subscription_id: number | null;
  purpose: string;
  provider: string;
  external_id: string | null;
  amount_kop: number;
  currency: string;
  status: string;
  created_at: string;
  paid_at: string | null;
}

export interface AdminPaymentsListParams {
  status?: string;
  provider?: string;
  purpose?: string;
  user_id?: number;
  created_from?: string;
  created_to?: string;
  q?: string;
  page?: number;
  page_size?: number;
}

export interface AdminPaymentsPage {
  items: AdminPaymentListItem[];
  total: number;
  page: number;
  page_size: number;
}
